cgsmiles/resolve.py

```diff
@@ -238,13 +238,12 @@ def resolve_disconnected_molecule(self, fragment_dict):
             a dict of fragment graphs
         """
         for meta_node in self.meta_graph.nodes:
-            neighbors = self.meta_graph.neighbors(meta_node)
-            edge_orders = [self.meta_graph.edges[(meta_node, ndx)]['order'] for ndx in neighbors]
-            # we are dealing with a virtual node that has no projection to the
-            # next resolution level
-            if edge_orders and all([order == 0 for order in edge_orders]):
-                continue
             fragname = self.meta_graph.nodes[meta_node]['fragname']
+
+            # we have a virtual node and skip it
+            if fragname not in fragment_dict:
+                continue
```

Comment:
Here you could check that it only has 0-order bonds, and issue a warning if not.

Reply:
done